src/oscar/apps/checkout/calculators.py

```diff
@@ -13,10 +15,18 @@ def __init__(self, request=None):
         # always changes the order total.
         self.request = request
 
-    def calculate(self, basket, shipping_charge, **kwargs):
-        excl_tax = basket.total_excl_tax + shipping_charge.excl_tax
-        if basket.is_tax_known and shipping_charge.is_tax_known:
-            incl_tax = basket.total_incl_tax + shipping_charge.incl_tax
+    def calculate(self, basket, shipping_charge, surcharges, **kwargs):
+        if surcharges is None:
```

Comment:
instead of creating a zero value surcharge why not move the surchage check below shipping charge check and add to excl_tax and incl_tax if required